### DISCUSSION TITLE
How to use API to upload my  photos to NocoDB?

### DISCUSSION THREAD

**Discussion by NeonBubbles** [nocodb/nocodb#6075] [Q&A] [answered]
I've spent the afternoon studying your API documentation but still couldn't understand how to upload attachments to the database. Even though the status code shows success, it doesn't seem to have actually worked.
1. I don't understand why it's necessary to provide information like file size and file name when uploading a document. Can't it just be uploaded directly?
2. How can I upload the file to a specific row in the database? The API documentation doesn't seem to explain this.
```
import requests



headers = {
    'xc-token': 'xxx',
}


file_path = r'D:\tg_photos\photo_2023-07-14_16-11-07.jpg'
file = open(file_path, 'rb')


data = {
    'mimetype': 'image/jpg',
    'path': 'download/noco/telegram_crawl/telergam_photos/photo_2023-07-14_16-11-07.jpg',
    'size': 12345, 
    'title': 'your_file_title',

}

response = requests.post(
    'http://xxx:8080/api/v1/db/storage/upload',
    headers=headers,
    files={'file': file},
    data=data,
)

print(response.text)


file.close()


if response.status_code == 200:
    print('File uploaded successfully')
else:
    print('Failed to upload file, status code:', response.status_code)


```

**Comment by dstala** [comment, marked as answer]
https://github.com/nocodb/nocodb/discussions/369

**Comment by NeonBubbles** [comment]
solution:
https://github.com/nocodb/nocodb/discussions/369#discussioncomment-6551312
